clients/python-wrapper/lakefs/object.py

```diff
@@ -2,19 +2,380 @@
+                pass
+
+        # fallback to ETag
+        etag = headers.get("ETag", "").strip(' "')
+        return etag
+
+    def _upload_raw(self, content: UploadContentType, content_type: Optional[str] = None,
+                    metadata: Optional[dict[str, str]] = None) -> ObjectStats:
+        """
+        Use raw upload API call to bypass validation of content parameter
+        """
+        headers = {
+            "Accept": "application/json",
+            "Content-Type": content_type if content_type is not None else "application/octet-stream"
+        }
+
+        # Create user metadata headers
+        if metadata is not None:
+            for k, v in metadata.items():
+                headers[_LAKEFS_METADATA_PREFIX + k] = v
+
+        _response_types_map = {
+            '201': "ObjectStats",
+            '400': "Error",
+            '401': "Error",
+            '403': "Error",
+            '404': "Error",
+            '412': "Error",
+            '420': None,
+        }
+        # authentication setting
+        _auth_settings = ['basic_auth', 'cookie_auth', 'oidc_auth', 'saml_auth', 'jwt_token']
+        return self._client.sdk_client.objects_api.api_client.call_api(
+            resource_path='/repositories/{repository}/branches/{branch}/objects',
+            method='POST',
+            path_params={"repository": self._repo_id, "branch": self._ref_id},
+            query_params={"path": self._path},
+            header_params=headers,
+            body=content,
+            response_types_map=_response_types_map,
+            auth_settings=_auth_settings,
+            _return_http_data_only=True
+        )
+
+    def _upload_presign(self, content: UploadContentType, content_type: Optional[str] = None,
+                        metadata: Optional[dict[str, str]] = None) -> ObjectStats:
+        headers = {
+            "Accept": "application/json",
+            "Content-Type": content_type if content_type is not None else "application/octet-stream"
+        }
+
+        staging_location = self._client.sdk_client.staging_api.get_physical_address(self._repo_id,
+                                                                                    self._ref_id,
+                                                                                    self._path,
+                                                                                    True)
+        url = staging_location.presigned_url
+        if self._client.storage_config.blockstore_type == "azure":
+            headers["x-ms-blob-type"] = "BlockBlob"
+
+        resp = self._client.sdk_client.objects_api.api_client.request(
+            method="PUT",
+            url=url,
+            body=content,
+            headers=headers
+        )
+
+        etag = WriteableObject._extract_etag_from_response(resp.getheaders())
+        staging_metadata = StagingMetadata(staging=staging_location,
+                                           size_bytes=len(content),
```

Comment:
len of content is always size bytes?

Reply:
Yes? Trying to understand whether this is a trick question... 🤔

Reply:
Gotcha, fixed and added test